### PR TITLE
Alternate fix for the SSL Issue

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

Alternate fix for issue  https://github.com/tardis-sn/carsus/issues/329.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: carsus/io/util.py [hunk 1]
```diff
@@ -106,3 +108,39 @@ def read_from_buffer(fname):
     checksum = hashlib.md5(buffer.getbuffer()).hexdigest()
 
     return buffer, checksum
+
+
+def retry_request(
+    url,
+    method,
+    retries=10,
```
Comment (s-rathi) [inline carsus/io/util.py:116]: As I mentioned in #330, hardcoding the retries to 10 may not be the best way to handle the issue.

**Review by s-rathi — CHANGES_REQUESTED** [review]

File: carsus/io/util.py [hunk 2]
```diff
@@ -106,3 +108,39 @@ def read_from_buffer(fname):
...
+):
+    """Retry an HTTP request.
+
+    Parameters
+    ----------
+    url : str
+        URL to send request to.
+    method : str
+        HTTP request method.
+    retries : int, default: 5, optional
```
Comment (chvogl) [inline carsus/io/util.py:129]:
```suggestion
    n_retry : int, default: 15, optional
```

File: carsus/io/util.py [hunk 3]
```diff
@@ -106,3 +108,39 @@ def read_from_buffer(fname):
...
+    backoff_factor : int, default: 1, optional
+    status_forcelist : list, default: [502, 503, 504, 400, 495], optional
+    **kwargs : dict, optional
+
+    Returns
+    -------
+    response: requests.Response
```
Comment (chvogl) [inline carsus/io/util.py:136]:
```suggestion
    response : requests.Response
```

**Review by chvogl — APPROVED** [review]
This looks good! I only noticed two small issues in one of the docstrings. We should decide soon which one of the two possible solutions we are going to merge @atharva-2001 @epassaro.

**Pull request merged**